src/main/kotlin/com/criptext/mail/push/data/PushAPIRequestHandler.kt

```diff
@@ -8,9 +8,20 @@ import android.support.annotation.RequiresApi
 import com.criptext.mail.R
 import com.criptext.mail.androidui.CriptextNotification
 import com.criptext.mail.api.HttpClient
+import com.criptext.mail.api.HttpErrorHandlingHelper
+import com.criptext.mail.db.AppDatabase
+import com.criptext.mail.db.EmailDetailLocalDB
+import com.criptext.mail.db.dao.EmailDao
 import com.criptext.mail.db.models.ActiveAccount
+import com.criptext.mail.db.models.EmailLabel
+import com.criptext.mail.db.models.Label
 import com.criptext.mail.push.PushTypes
+import com.criptext.mail.scenes.emaildetail.data.EmailDetailResult
+import com.criptext.mail.scenes.label_chooser.SelectedLabels
+import com.criptext.mail.scenes.label_chooser.data.LabelWrapper
 import com.github.kittinunf.result.Result
+import com.github.kittinunf.result.flatMap
+import com.github.kittinunf.result.mapError
 import org.json.JSONObject
 
 class PushAPIRequestHandler(private val ctx: Context, private val manager: NotificationManager){
```

Comment:
Why do you need a `Context` reference? Don't use `NotificationManager` directly, you should use an abstraction over it.